lib/Models/CatalogItem.js

```diff
@@ -984,7 +999,14 @@ function isEnabledChanged(catalogItem) {
             catalogItem._loadForEnablePromise = resolvedOrRejected ? undefined : loadPromise;
         }
 
-        catalogItem.isShown = true;
+        if (catalogItem.isShown) {
+            // If item already had the isShown flag but wasn't enabled, it wasn't actually shown, so call isShownChanged
+            // to update the view.
```

Comment:
When would this happen?  Setting `isShown` will set `isEnabled` first and then wait for the enabling to happen for continuing with the showing.  It seems like adding this logic here could result in the showing logic happening twice in some cases?